js/ui/keyboard.ts

```diff
@@ -460,24 +508,151 @@ export default class KeyBoard {
+                        break;
+                    case 'Т':
+                        key = 't';
+                        break;
+                    case 'Ъ':
+                        key = 'y';
+                        break;
+                    case 'У':
+                        key = 'u';
+                        break;
+                    case 'И':
+                        key = 'i';
+                        break;
+                    case 'О':
+                        key = 'o';
+                        break;
+                    case 'П':
+                        key = 'p';
+                        break;
+                    case 'Ю':
+                        key = '@';
+                        break;
+                    case '@':
+                        key = '`';
+                        break;
+
+                    // Third row.
+                    case 'А':
+                        key = 'a';
+                        break;
+                    case 'С':
+                        key = 's';
+                        break;
+                    case 'Д':
+                        key = 'd';
+                        break;
+                    case 'Ф':
+                        key = 'f';
+                        break;
+                    case 'Г':
+                        key = 'g';
+                        break;
+                    case 'Х':
+                        key = 'h';
+                        break;
+                    case 'Й':
+                        key = 'j';
+                        break;
+                    case 'К':
+                        key = 'k';
+                        break;
+                    case 'Л':
+                        key = 'l';
+                        break;
+                    case 'Ш':
+                        key = '[';
+                        break;
+                    case 'Щ':
+                        key = ']';
+                        break;
+                    case '[':
+                        key = '{';
+                        break;
+                    case ']':
+                        key = '}';
+                        break;
+
+                    // Fourth row.
+                    case 'З':
+                        key = 'z';
+                        break;
+                    case 'Ь':
+                        key = 'x';
+                        break;
+                    case 'Ц':
+                        key = 'c';
+                        break;
+                    case 'Ж':
+                        key = 'v';
+                        break;
+                    case 'Б':
+                        key = 'b';
+                        break;
+                    case 'Н':
+                        key = 'n';
+                        break;
+                    case 'М':
+                        key = 'm';
+                        break;
+
+                    default:
+                        break;
+                }
+            }
+
```

Comment:
This makes me think that I probably should have used a table lookup for this before, but that's a future optimization.